### PR TITLE
Added cpp versions of sysid routines and example

### PR DESCRIPTION
**Pull request by r4stered**

### CODE REVIEW

File: wpilibcExamples/src/main/cpp/examples/SysId/include/subsystems/Drive.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      frc2::sysid::Config{std::nullopt, std::nullopt, std::nullopt,
+                          std::nullopt},
+      frc2::sysid::Mechanism{
+          [this](units::volt_t driveVoltage) {
+            m_leftMotors.SetVoltage(driveVoltage);
+            m_rightMotors.SetVoltage(driveVoltage);
+          },
+          [this](frc::sysid::MotorLog* log) {
+            log->RecordFrameLinear(
+                m_appliedVoltage = m_leftMotors.Get() *
```
Comment (Oblarg) [inline wpilibcExamples/src/main/cpp/examples/SysId/include/subsystems/Drive.h]: Should add a `getVoltage` method to the CPP-side `MotorController` class that returns in the right units

Comment (calcmogul) [reply wpilibcExamples/src/main/cpp/examples/SysId/include/subsystems/Drive.h]: That's out of scope for this PR, so I opened an issue: https://github.com/wpilibsuite/allwpilib/issues/6043

**Comment by calcmogul** [comment]
Looks like you messed up your rebase.

**Comment by r4stered** [comment]
Yeah, I got my repo in a bad state and ended up having to reclone... I am working on adding the unit tests right now. Should be done today hopefully.

**Comment by r4stered** [comment]
Should have unit tests now!

**Comment by calcmogul** [comment]
The C++ tests are crashing.

**Comment by r4stered** [comment]
> The C++ tests are crashing.

Thats odd. Once I get some time I can try to reproduce the crash. Been busy with work lately....

**Pull request closed without merging**